### PR TITLE
[loss refactor] [1] add snapshot test on loss functions

### PR DESCRIPTION
**Pull request by yueming-yuan**

### CODE REVIEW

File: tests/fast/backends/training_utils/loss/loss_test_utils.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+import copy
+import subprocess
+from argparse import Namespace
+from pathlib import Path
+
+import torch
+
+from miles.backends.training_utils.parallel import ParallelState
+
+ARTIFACTS_REPO = "https://github.com/yueming-yuan/miles-artifacts.git"
```
Comment (Shi-Dong) [inline tests/fast/backends/training_utils/loss/loss_test_utils.py:33]: Would it be better to use a repo under `radixark`?

File: tests/fast/backends/training_utils/loss/test_loss_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    get_values,
+    icepop_function,
+    loss_function,
+    policy_loss_function,
+    sft_loss_function,
+    value_loss_function,
+    vanilla_tis_function,
+)
+
+from .loss_test_utils import (
```
Comment (Shi-Dong) [inline tests/fast/backends/training_utils/loss/test_loss_snapshot.py:34]: I'd suggest using absolute import (since `__init__.py` exists under `test/` so it is a proper package).

File: tests/fast/backends/training_utils/loss/README.md
Comment (Shi-Dong) [inline tests/fast/backends/training_utils/loss/README.md:1]: I think it'd be great to specify in this readme
1. Which loss functions are covered by `test_loss_snapshot.py`;
2. What "new config" means.

File: tests/fast/backends/training_utils/loss/test_loss_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    return "compare"
+
+
+# ---------------------------------------------------------------------------
+# Runners: call each function and collect outputs
+# ---------------------------------------------------------------------------
+
+
+def _get_sum_of_sample_mean(batch, args, parallel_state):
+    from miles.backends.training_utils.cp_utils import get_sum_of_sample_mean
```
Comment (Shi-Dong) [inline tests/fast/backends/training_utils/loss/test_loss_snapshot.py:128]: Move import to the top?

**Review by Shi-Dong — APPROVED** [review]
LGTM. All comments are style-wise.

**Comment by yueming-yuan** [comment]
Closing without merging — the snapshot test infrastructure landed here is fully contained in #753's history (PR1's tip is an ancestor of PR2's tip). To avoid the surgical force-push needed to split the two cleanly, we'll just land the merged work as part of #753 → #1121 → #1132. Approval here is appreciated; the same content will be reviewable in the diffs of the downstream PRs.
